src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -35,6 +38,9 @@ Q_LOGGING_CATEGORY(lcCfApiWrapper, "nextcloud.sync.vfs.cfapi.wrapper", QtInfoMsg
       FIELD_SIZE( CF_OPERATION_PARAMETERS, field ) )
 
 namespace {
+using SyncRootKeys = QMap<QString, QString>;
+Q_GLOBAL_STATIC(SyncRootKeys, registeredSyncRootKeys)
+QReadWriteLock registeredSyncRootKeysLock;
```

Comment:
Do we have so much contention on it that we need a read/write lock? It's much slower than a mutex in the general case.

Reply:
@er-vin Fixed. Using a mutex now. But, I've got another idea. At the moment, we are always registering sync folders on application launch. Ideally, we shouldn't be doing this part if it is done already in the previous run, and we are not adding any new folders. So, I guess, I shouldn't really keep track of folders that I have added in the runtime, as, if we change this part in the future, I won't have information of registered folders if I restart the app. So, I guess, I need a way to traverse all the keys under **SyncRootManager\** and check those that match by _providername!WindowsSID!accountDisplayName_ and then just check if _UserSyncRoots_ has a key that has a value of the sync root path we are removing. This will allow me to ditch this static map and also make the logic more reliable and less likely to becoe broken. Will just need to somehow pass the **providerName** and **userDisplayName** into unregister function, somehow.

Reply:
@er-vin I've removed that global static completely.